base/set.jl

```diff
@@ -422,3 +422,28 @@ end
 
 convert(::Type{Set{T}}, s::Set{T}) where {T} = s
 convert(::Type{Set{T}}, x::Set) where {T} = Set{T}(x)
+
+# replace!
+
+askey(k, ::Associative) = k.first
+askey(k, ::AbstractSet) = k
+
+function _replace!(prednew::Callable, A::Union{Associative,AbstractSet}, count::Int)
+    count < 0 && throw(DomainError())
```

Comment:
Done, but I see that some `@noinline` were added by Tim in his PR, should I bother with that here?